### DISCUSSION TITLE
DNS caching?

### DISCUSSION THREAD

**Discussion by jwodder** [encode/httpx#2211] [Q&A] [answered]
What is the nature of httpx's DNS caching, if any?  I'm currently working with a script that creates multiple concurrent `AsyncClient`s connecting to the same server, and I'm running into DNS errors (specifically, `socket.gaierror`, "Nodename nor servname provided or not known") that go away when the number of concurrent clients is heavily limited.  The ability to cache DNS lookups at a super-client level in order to avoid this problem would be helpful.

**Comment by lovelydinosaur** [comment, marked as answer]
The package does not currently provide any DNS caching.

For comparison with other similar package, `aiohttp` does, but `requests` does not.

Agree that it's probably worthwhile for us to have on the roadmap. I'd be happy for someone to open an issue to request it as a feature.
